### DISCUSSION TITLE
Batch generation of English translated subtitles of uploaded mp3 files in Google Drive?

### DISCUSSION THREAD

**Discussion by archz2** [openai/whisper#1241] [Q&A]
I have lots of mp3 files in chinese language extracted from videos. 
I am currently extracting english subtitles from them one by one. I upload them to google drive and then drag in the open folder inside the collab notebook. 

I use the following code. 

```
import whisper

model = whisper.load_model("large")

!whisper nameofmyfile.mp3 --task translate
```

How can I execute a batch command in such a way that after one mp3 files get complete, the next  one begins and the srt files get saved in a separate folder in my google drive?
![image](https://user-images.githubusercontent.com/66735186/232322945-070127fb-ad03-41b1-920e-f11f830c31f6.png)

The path of the uploaded mp3 files will be as follows.

Mydrive/audiostuff/thefirstone.mp3

Mydrive/audiostuff/thesecondone.mp3

The output path should be
Mydrive/extractedstuff/thesecondone.srt  , etc.

**Comment by phineas-pta** [comment]
you write a `for` loop

**Comment by archz2** [comment]
Please share a step by step method. I'm a noob.

**Reply by phineas-pta** [reply]
`for` loop is one of the most basic things in writing code, if you know nothing about python then you should use a graphical interface, for e.g. https://github.com/tigros/Whisperer

**Reply by wenchaoliu-93** [reply]
There are a few ways.  If you use a Linux machine, you can easily achieve that on the command line. You can also implement this in code, presumably Python. Either way, it seems easiest if audio files have been downloaded, so there'd not be any overhead in interfacing with Google Drive.
Alternatively, use Google or AWS: https://groups.google.com/g/cloud-speech-discuss/c/jWdh5mK93K4/m/VFpmWehtCAAJ?utm_medium=email&utm_source=footer
